doc/source/whatsnew/v3.0.0.rst

```diff
@@ -118,7 +118,7 @@ Bug fixes
 ~~~~~~~~~
 - Fixed bug in :meth:`DataFrame.join` inconsistently setting result index name (:issue:`55815`)
 - Fixed bug in :meth:`Series.diff` allowing non-integer values for the ``periods`` argument. (:issue:`56607`)
--
+- Fixed bug in :meth:`api.types.is_datetime64_any_dtype` where a custom :class:`ExtensionDtype` would not return ``True`` for array-likes (:issue:`57055`)
```

Comment:
Could you put this under the `ExtensionArray` section

Reply:
Sure. Should I add that it now returns `False` for pyarrow dates?